### ISSUE TITLE
Show overview of pending changes of repo on the dashboard

### ISSUE THREAD

**Issue by dotnetjunkie** [gitextensions/gitextensions#8356]
For some time now, GitExtensions starts with showing a dashboard, which shows a list of all local repositories. Considering the last list of repositories in my dashboard, I often have trouble finding the repository that I just made changes to.

It might be useful to add some visual clues to the dashboard that help spotting repositories that have changes. For instance, adding the number of pending changes with a color indicator would be very helpful. Example:

![image](https://user-images.githubusercontent.com/1506621/88475980-fee6a980-cf34-11ea-966b-946fda7242eb.png)

**Comment by vbjay** [comment]
The problem is when you have a lot of repos in the dashboard, this would
bring it to it's knees. It had to run a git command to get this
information.

**Comment by gerhardol** [comment]
The git-status command required can take anywhere from 100 ms in small repos to 30s for the initial run in larger repos with viruses like Symantec, so this can be slow. GE would spin up at least one CPU for quite some time  if you have more than a few repos, so users will likely complain about GE using a lot of resources at startup.

**Comment by dotnetjunkie** [comment]
Would it be possible to do this asynchronously (in the background)? This of course doesn't prevent the CPU spike, but still makes the UI responsive.

**Comment by RussKie** [comment]
Yes, it can be done the same way branches are shown in the dashboard.
Pull requests are welcome.

**Comment by vbjay** [comment]
The other issue. I open a repo while it's doing this processing. We will
have fun with index locks.

**Comment by RussKie** [comment]
> The other issue. I open a repo while it's doing this processing. We will have fun with index locks.

Background operations will have to be cancelled as soon as the dashboard is closed.

**Comment by vbjay** [comment]
Cancelation only covers it to a point. It started a git process and then I
open the repo. The repo opening does the whole read the info thing and runs
a ton of git commands. The task that was doing this repo work may have been
canceled but the git process is still running. I guess the big question is
what happens when you run multiple git status calls and the other calls
like git branch and whatnot during repo opening at the same time. Does git
set an index lock at all during repo open process.

**Comment by gerhardol** [comment]
GE runs git-status for the commit count with 
https://git-scm.com/docs/git#Documentation/git.txt---no-optional-locks
This will not set any locks so interactive commands have precedence (but also that caches are not updated, the status checks can run slow).
The same option is added for other commands that run in the background (but last time I checked it was just implemented for git-status).

So this could run async one by one repo and be cancelled after user select the repo to use. But the update rate in the GUI will be slow and one CPU core will be fully occupied for some time...
